server/enterprise/metrics/histogram.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.enterprise for license information.
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+)
+
+// HistogramVec is a wrapper of prometheus.HistogramVec that stores the buckets
+// which is later passed down to WrappedObserver.
+type HistogramVec struct {
+	*prometheus.HistogramVec
+	buckets []float64
+	fqName  string
+	logger  mlog.LoggerIFace
+}
+
+// WrappedObserver is a wrapper of prometheus.Observer which also addtionally
+// logs userIDs when the value exceeds or equal to the highest bucket.
```

Comment:
Nit: other typo and wording
```suggestion
// WrappedObserver is a wrapper of prometheus.Observer which additionally
// logs userIDs when the value exceeds or equals the one in the highest bucket.
```